quic/s2n-quic-tls/src/session.rs

```diff
@@ -24,9 +24,6 @@ pub struct Session {
     state: callback::State,
     handshake_complete: bool,
     send_buffer: BytesMut,
-    emitted_server_name: bool,
-    // This is only set for the client to avoid an extra allocation
-    server_name: Option<ServerName>,
```

Comment:
Ok so this is why we had this functionality... The client knows the server name before the session even starts and it's wasteful to pass that to s2n-tls and then have to allocate/copy the value back out again. Maybe we should support this for the client and the change you made for the server instead?

Reply:
Not sure I'm following, are you saying that we want to keep the server_name field in this struct?

Reply:
Correct. But just for the client.